gplugins/tidy3d/component.py

```diff
@@ -498,6 +499,7 @@ def write_sparameters(
         pad_z_outer=pad_z_outer,
         dilation=dilation,
     )
+    path_dir = str(pathlib.Path(dirpath) / str(hash(c)))
```

Comment:
This will hash to different values on different runs I believe? If we use the fix from flexcompute/tidy3d#1259 then we should call `c._hash_self()` here instead.

Reply:
it has the same value on different runs

Reply:
are you sure? across different interpreter instances?

Reply:
the hash is coming from Tidy3D component modeller

i run with different interpreter instances and was working well

Reply:
ok perfect then :+1:

Reply:
Hi Tyler, yes, but `Tidy3DComponent` is not your modeler but the `LayeredComponentBase`


```
class Tidy3DComponent(LayeredComponentBase):
    """
    Represents a 3D component in the Tidy3D simulation environment.

    Attributes:
        material_mapping (dict[str, Tidy3DMedium]): A mapping of material names to Tidy3DMedium instances.
        extend_ports (NonNegativeFloat): The extension length for ports.
        port_offset (float): The offset for ports.
        pad_xy_inner (NonNegativeFloat): The inner padding in the xy-plane.
        pad_xy_outer (NonNegativeFloat): The outer padding in the xy-plane.
        pad_z_inner (float): The inner padding in the z-direction.
        pad_z_outer (NonNegativeFloat): The outer padding in the z-direction.
        dilation (float): Dilation of the polygon in the base by shifting each edge along its
            normal outwards direction by a distance;
            a negative value corresponds to erosion. Defaults to zero.
    """

    material_mapping: dict[str, Tidy3DMedium] = material_name_to_medium
    extend_ports: NonNegativeFloat = 0.5
    port_offset: float = 0.2
    pad_xy_inner: NonNegativeFloat = 2.0
    pad_xy_outer: NonNegativeFloat = 2.0
    pad_z_inner: float = 0.0
    pad_z_outer: NonNegativeFloat = 0.0
    dilation: float = 0.0

```

Reply:
oh. my fault, please disregard